scripts/source-maintenance.sh

```diff
@@ -408,8 +447,14 @@ run_ checkMakeNamedErrorDetails || exit 1
 # Run formatting
 echo "" >doc/debug-sections.tmp
 srcFormat || exit 1
-sort -u <doc/debug-sections.tmp | sort -n >doc/debug-sections.tmp2
-cat scripts/boilerplate.h doc/debug-sections.tmp2 >doc/debug-sections.txt
+
+if [ -z "$OnlyChangedSince" ]; then
+    sort -u <doc/debug-sections.tmp | sort -n >doc/debug-sections.tmp2
+    cat scripts/boilerplate.h doc/debug-sections.tmp2 >doc/debug-sections.txt
+else
+    echo "--only-changed-since specified, Skipping updating doc/debug-sections.txt"
+fi
+
 rm doc/debug-sections.tmp doc/debug-sections.tmp2
```

Comment:
Is this rm command not complaining about "no such file" for the *.tmp2 removal ?

Reply:
No. What happens is that the debug-sections.txt file is only populated with sections from the files that have been scanned

Reply:
Yes, current PR changes may result in bogus doc/debug-sections.tmp contents that should not be used, but Amos' concern is still valid: Current PR changes may prevent `doc/debug-sections.tmp2` creation in some cases. The `rm` command will produce a mysterious (to the developer running the script) error message in those cases.

Reply:
right, changed that